### ISSUE TITLE
Can't refresh token via Ping Identity OIDC

### ISSUE THREAD

**Issue by liquid-cloud** [goharbor/harbor#9267]
If you are reporting a problem, please make sure the following information are provided:

**Expected behavior and actual behavior:**
Logging in to Harbor via Ping Identity OIDC. Expecting harbor to be able to automatically refresh the token and maintain the session. Currently, harbor fails to refresh the token and the session expires after 5 minutes (which is when `id_token` expires for the first time and the first refresh needs to happen).

**Steps to reproduce the problem:**
Steps:
1. Login using the OIDC button
2. Wait 5 minutes (`id_token` expires after that)
3. Session is terminated, user is asked to login back.

Logs from harbor-core:
```
2019-09-26T07:29:41Z [DEBUG] [/core/filter/security.go:445]: got user information from session
2019-09-26T07:29:41Z [DEBUG] [/common/utils/oidc/secret.go:105]: Token string for verify: {"access_token":"0001sFj1DVx09st4rTkg4F5yEp5g","token_type":"Bearer","refresh_token":"O9bESXK6RWi7hue4KaFvzcGkpOJz4dYBmEQuQUfohf","expiry":"2019-09-26T09:24:34.412428487Z","id_token":"<long_id_token>"}
2019-09-26T07:29:41Z [INFO] [/common/utils/oidc/secret.go:110]: Failed to verify ID Token, error: oidc: token is expired (Token Expiry: 2019-09-26 07:29:32 +0000 UTC), refreshing...
```
As you can see, the `access_token` is not expired yet, but harbor is looking at `id_token` which expires after 5 minutes. A new `id_token` can be requested from the PingIdentity OIDC when it expires using the refresh method.

I've tried investigating what happens, and found out that probably the refresh call never happens. Harbor uses `TokenSource` [here in the code](https://github.com/goharbor/harbor/blob/master/src/common/utils/oidc/helper.go#L198) which returns the `Token t` until that expires according to the [documentation](https://godoc.org/golang.org/x/oauth2#Config.TokenSource).

I've added an extra line for logging, which shows that it uses the `access_token` which is not yet expired:
```
2019-09-26T07:29:41Z [ERROR] [/common/utils/oidc/helper.go:199]: === Using token.Token to refresh: &{AccessToken:0001sFj1DVx09st4rTkg4F5yEp5g TokenType:Bearer RefreshToken:O9bESXK6RWi7hue4KaFvzcGkpOJz4dYBmEQuQUfohf Expiry:2019-09-26 09:24:34.412428487 +0000 UTC raw:<nil>}
```
Which is not expired. So the `TokenSource` method will return the exact same token back. This means it will not contain the `id_token` which Harbor is looking for. It will then fail [here](https://github.com/goharbor/harbor/blob/master/src/common/utils/oidc/helper.go#L205) and the session gets invalidated asking the user to login again.

**Versions:**
Please specify the versions of following systems.

- harbor version: 1.9.0
- docker engine version: 18.6.3
- docker-compose version: [z.z.z]
- helm chart version: 1.2.0

**Additional context:**

Might be related to issue #8069.

- **Harbor config files:** You can get them by packaging `harbor.yml` and files in the same directory, including subdirectory.
- **Log files:** You can get them by package the `/var/log/harbor/` .

**Comment by reasonerjt** [comment]
@liquid-cloud 
Thanks for reporting this, consolidating the information in this issue and #8096 it seems to me it's correct not to expect id token in the response of token refresh

**Comment by reasonerjt** [comment]
Fixed in #9481 #9435

**Comment by liquid-cloud** [comment]
I just checked the code with the above commits and it works fine (branch `release-1.9.0`).

The `RefreshToken` call was successful and a new `access_token` was returned. Tested using the CLI.
